icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -196,6 +236,25 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         if (bestSnlp[numCodePts] == kint32max) {
             t_boundary[numBreaks] = numCodePts;
             numBreaks++;
+        } else if (isPhraseBreaking) {
+            t_boundary[numBreaks] = numCodePts;
+            numBreaks++;
+            int prevIdx = numCodePts;
+            int codeUnitIdx = 0, length = 0;
+            for (int i = prev[numCodePts]; i > 0; i = prev[i]) {
+                codeUnitIdx = prenormstr.offsetByCodePoints(0, i);
+                length = prevIdx - i;
+                prevIdx = i;
+                String pattern = getPatternFromText(text, s, codeUnitIdx, length);
+                // Keep the breakpoint if the pattern is not in the fSkipSet and continuous Katakana
+                // characters don't occur.
+                text.setIndex(codeUnitIdx - 1);
```

Comment:
I am going to approve again, but could you take another look at this line and its `- 1`?
It looks like that is intended to "go one *code point* before codeUnitIdx" (see the following use of current32() and next32()) which could be one or two code units.

Also, is it possible here that codeUnitIdx == 0?

Reply:
I asked the same question in the C++ review and Allen said line 244 and 245 will ensure codeUnitIdx > 0 always

Reply:
Ok. What about one code point vs. one code unit?

Reply:
I made some modification here. Test case will be added soon.
https://github.com/unicode-org/icu/pull/1965